### PR TITLE
Sunken Treasure

### PR DESCRIPTION
**Pull request by BluePsychoRanger**
Blue has stated that Sunken Treasures was discontinued because MC added buried treasure, but I don't see why this completely different module has to be scrapped. In my opinion, this module is more like sand sifters than buried treasure found from exploration maps.

Percentages are the same as the 1.12 version, but sponge was replaced with a treasure chest with a loot table containing a single treasure item -- 1-2 wet sponge, trident with 0-2 durability, OR heart of the sea -- along with other items.

EDIT: sponge chance is now 1.5%; treasure chest chance now is 0.5%

### CODE REVIEW

**Comment by BluePsychoRanger** [comment]
In the treasure chests, there is a 81.97% chance of getting 1-2 wet sponges, a 16.39% chance of getting a heart of the sea, and a 1.64% chance of getting a trident with 0-2 durability. A treasure chest has a 0.5% chance to spawn, so the chances per treasure spawn is 0.41%, 0.082%, and 0.0082% respectively. When using a fortune III tool, per sand block mined, the percentages are 0.103%, 0.021%, and 0.0021% respectively.

TL;DR the chance to get a trident is very low, and even then, the durability is very low as well.

**Comment by misode** [comment]
@BluePsychoRanger Yes, there is a builtin way using a different advancement criteria for each item
```json
"criteria":{
  "my_audrey_1":{
    "trigger":"minecraft:inventory_changed",
    "conditions": {
      "items": [
        {
          "item": "minecraft:glass_bottle",
          "nbt": "{gm4_audrey:1b}"
        }
      ]
    }
  },
  "my_audrey_2":{
    "trigger":"minecraft:inventory_changed",
    "conditions": {
      "items": [
        {
          "item": "minecraft:glass_bottle",
          "nbt": "{gm4_audrey:2b}"
        }
      ]
    }
  },
  ...
}
```

**Comment by BluePsychoRanger** [comment]
If players cannot, then I guess it could instead place the chest at the location of the sand block. I can't test it though.
